merkle/map_hasher.go

```diff
@@ -1,40 +0,0 @@
-// Copyright 2016 Google Inc. All Rights Reserved.
-//
-// Licensed under the Apache License, Version 2.0 (the "License");
-// you may not use this file except in compliance with the License.
-// You may obtain a copy of the License at
-//
-//     http://www.apache.org/licenses/LICENSE-2.0
-//
-// Unless required by applicable law or agreed to in writing, software
-// distributed under the License is distributed on an "AS IS" BASIS,
-// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-// See the License for the specific language governing permissions and
-// limitations under the License.
-
-package merkle
```

Comment:
Putting this code in it's own package
- Supports simple Go style instantiation: `maphasher.New()` 
- Creates a smaller package, which is easier for testing and dependency tracking. 
- Gives us the option of making the `type MapHasher struct` private (`type mapHasher struct`), in order to force referrers to only refer to the object through it's interface (`merkle.TreeHasher`)